scripts/extractions/extract_common.py

```diff
@@ -0,0 +1,191 @@
+    job_items: List[pystac.Item], row: pd.Series, parameters: dict = {}
+) -> list:
+    """From the job items, extract the metadata and save it in a netcdf file."""
+    base_gpd = gpd.GeoDataFrame.from_features(json.loads(row.geometry)).set_crs(
+        epsg=4326
+    )
+    assert len(base_gpd[base_gpd.extract == 2]) == len(
+        job_items
+    ), "The number of result paths should be the same as the number of geometries"
+
+    extracted_gpd = base_gpd[base_gpd.extract == 2].reset_index(drop=True)
+    # In this case we want to burn the metadata in a new file in the same folder as the S2 product
+    for idx, item in enumerate(job_items):
+        if "sample_id" in extracted_gpd.columns:
+            sample_id = extracted_gpd.iloc[idx].sample_id
+        else:
+            sample_id = extracted_gpd.iloc[idx].sampleID
+
+        ref_id = extracted_gpd.iloc[idx].ref_id
+        valid_time = extracted_gpd.iloc[idx].valid_time
+        h3_l3_cell = extracted_gpd.iloc[idx].h3_l3_cell
+        s2_tile = row.s2_tile
+
+        item_asset_path = Path(list(item.assets.values())[0].href)
+
+        # Add some metadata to the result_df netcdf file
+        new_attributes = {
+            "start_date": row.start_date,
+            "end_date": row.end_date,
+            "valid_time": valid_time,
+            "GFMAP_version": version("openeo_gfmap"),
+            "creation_date": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+            "description": parameters.get(
+                "description", "Sentinel1 GRD raw observations, unprocessed."
+            ),
+            "title": parameters.get("title", "Sentinel-1 GRD"),
+            "sample_id": sample_id,
+            "ref_id": ref_id,
+            "spatial_resolution": parameters.get("spatial_resolution", "20m"),
+            "s2_tile": s2_tile,
+            "h3_l3_cell": h3_l3_cell,
+        }
+
+        # Saves the new attributes in the netcdf file
+        update_nc_attributes(item_asset_path, new_attributes)
+
+    return job_items
+
+
+def generate_output_path(
+    root_folder: Path, geometry_index: int, row: pd.Series, s2_grid: gpd.GeoDataFrame
+):
+    """Generate the output path for the extracted data, from a base path and
+    the row information.
+    """
+    features = geojson.loads(row.geometry)
+    sample_id = features[geometry_index].properties.get("sample_id", None)
+    if sample_id is None:
+        sample_id = features[geometry_index].properties["sampleID"]
+    ref_id = features[geometry_index].properties["ref_id"]
+
+    if "orbit_state" in row:
+        orbit_state = f"_{row.orbit_state}"
+    else:
+        orbit_state = ""
+
+    s2_tile_id = row.s2_tile
+    h3_l3_cell = row.h3_l3_cell
+    epsg = s2_grid[s2_grid.tile == s2_tile_id].iloc[0].epsg
+
+    subfolder = root_folder / ref_id / h3_l3_cell / sample_id
+    return (
+        subfolder
+        / f"{row.out_prefix}{orbit_state}_{sample_id}_{epsg}_{row.start_date}_{row.end_date}{row.out_extension}"
```

Comment:
What is `orbit_state` for non-S1 extractions?

Reply:
by default the value is an empty string  `""` so it doesn't impact the file name if there is no orbit information in the `row: pd.Series` variable